### ISSUE TITLE
Syntax coloring is wrong with new code navigation beta

### ISSUE THREAD

**Issue by jack1142** [StylishThemes/GitHub-Dark#1052]
* **Browser**: Chrome
* **Operating System**: Windows 10
* **Link to page with the issue**: https://github.com/mikeshardmind/SinbadCogs/blob/v3/antimentionspam/antimentionspam.py
* **Screenshot**:
GH did some changes to syntax coloring style and now syntax coloring looks badly on GH Dark, and it doesn't resemble how Monokai theme looks in code editors at all :(
How it looks on GH Dark
![chrome_2020-01-11_13-53-44](https://user-images.githubusercontent.com/6032823/72204568-cfc98880-3479-11ea-8760-2a6a49f08d11.png)
How it looks on regular GH
![chrome_2020-01-11_13-49-04](https://user-images.githubusercontent.com/6032823/72204574-e839a300-3479-11ea-8ae0-7251f7dca4e7.png)

How it looked before on GH Dark (without code navigation, you can see it when you're not logged in):
![opera_2020-01-11_14-10-31](https://user-images.githubusercontent.com/6032823/72204798-259f3000-347c-11ea-8e84-a7f89e07e6be.png)
How it looked before on regular GH (without code navigation, you can see it when you're not logged in):
![image](https://user-images.githubusercontent.com/6032823/72204800-3354b580-347c-11ea-9106-ad0109e1f663.png)

**Comment by xt0rted** [comment]
For anyone looking into this, this looks to be `Monokai - Spacegray  Eighties` not the regular `Monokai` theme.

**Comment by jack1142** [comment]
Yeah, sorry for not saying that, but the new code navigation beta makes all Syntax coloring themes look wrong anyway.

**Comment by the-j0k3r** [comment]
> Yeah, sorry for not saying that, but the new code navigation beta makes all Syntax coloring themes look wrong anyway.

Well thats not exactly good news =(, we have umpteen themes and could use the help in this. :+1: :heart:

**Comment by xt0rted** [comment]
It looks like they're adding classes when code navigation is enabled. On the first line of that code file the `annotations` text isn't wrapped in a span, so it's getting the default text color from the theme. But with code navigation enabled it's being wrapped in `.pl-s1`. If we try to fix this we'll probably just make things worse elsewhere.

**Comment by jack1142** [comment]
We could also argue that the whole syntax coloring with code navigation enabled and GH Dark not installed looks wrong too, so it's actually an issue on GitHub's side, I mean, the new syntax coloring is just wrong in general...
This is how part of the code looks with code navigation enabled:
![chrome_2020-01-11_18-19-32](https://user-images.githubusercontent.com/6032823/72208072-f77f1780-349e-11ea-92c7-588e00713f25.png)
And this is how it looks with it disabled:
![chrome_2020-01-11_18-19-39](https://user-images.githubusercontent.com/6032823/72208081-12ea2280-349f-11ea-934a-081fb7930839.png)

**Comment by the-j0k3r** [comment]
How can we compete, I ask with those issues in place? But since you mentioned this was a beta, them I would suggest you file that bug upstream or send them feedback, GitHub should know about this at this beta stage.

With that taken care of then we can revisit this when feature is more mature.

**Comment by silverwind** [comment]
I too noticed `.pl-s1` being used a lot more with navigation enabled. That class is colored in many of our themes but neutral in GH's theme so they probably didn't even notice the issue. Maybe we should just remove coloring from that one class in all themes.

In general, our themes aren't that great and could be improved significantly. I wonder if there is a repo of themes and their colors that could be used to programmatically build themes instead of the current manual approach.

**Comment by the-j0k3r** [comment]
> In general, our themes aren't that great and could be improved significantly. I wonder if there is a repo of themes and their colors that could be used to programmatically build themes instead of the current manual approach.

yep and yep. But I doubt theres such a repo with the themes we use and to actually do it programmatically some 10+ themes would require some special magic and a good commitment to start/finish that job, I'll help where I can when I can but this is outside my current skills with JS.

> That class is colored in many of our themes but neutral in GH's theme so they probably didn't even notice the issue. Maybe we should just remove coloring from that one class in all themes.

I disagree with this route, we shouldn't be adapting our already not great themes to beta/production buggy upstream features. These bugs need to be reported upstream and it should be fixed upstream. Then we can deal with whatever.

**Comment by silverwind** [comment]
> some 10+ themes would require some special magic

I'm happy to reduce the amount of themes to a handful of well-supported ones. Quality over Quantity.

**Comment by the-j0k3r** [comment]
>I'm happy to reduce the amount of themes to a handful of well-supported ones. Quality over quantity.

We have 8 themes that have all counter parts GitHub | CodeMirror | Jupyter , so that would be a good place to start, though at any rate to generate themes for those alone still no easy task in automation.

**Comment by FichteFoll** [comment]
Can I disable code navigation on the client (i.e. my) side or is this something Github just rolled out for selected repositories and that is now active for all repos? I don't really care about code navigation, but I do care about proper code highlighting.

**Comment by xt0rted** [comment]
@FichteFoll the [changelog](https://github.blog/?s=code+navigation) posts on the blog will show when this was enabled for the various languages (more on the [help site](https://help.github.com/en/github/managing-files-in-a-repository/navigating-code-on-github)). There's currently no way to disable it either.

**Comment by the-j0k3r** [comment]
@lee-dohm would a report via contact with a link to this issue be OK with you?

**Comment by silverwind** [comment]
The green text in Twilight also bothers me a tiny bit, maybe I'll check it out later.

**Comment by lee-dohm** [comment]
@the-j0k3r Sorry, I'm just coming back from a two-week vacation and I'm still catching up 😀 If you're asking if it's ok with me that you contact GitHub Support to report an issue with the new code navigation beta, then, in general, yes that's completely fine with me 🙂 If I'm missing some nuance to the situation ... could you be more specific? (Or wait for me to get caught up on all the other things I've got swirling around my head right now and then I'll circle back 😆)

**Comment by jack1142** [comment]
@silverwind I think this only fixes the issue mentioned in #1132, not the whole issue with syntax coloring that happened due to GitHub's new code navigation feature.

**Comment by auscompgeek** [comment]
Related: #256

**Comment by the-j0k3r** [comment]
If the issue isnt fixed upstream, and the indication is thats this is the new normal, since, for codemirror, e.g. twilight you need 35 colors, the new CSS vars are only 19 at this time.

Their prettylights CSS vars are 30 strong so this is the closest we get unless we add more vars in the modding world of themes, at some stage it maybe better just todo our own thing for greater quality or try to integrate with their new vars system and be left wanting for some themes.

I have a POC for syntax theme switcher based on the latter only twilight working across all required stylesheets, but needs more work before its released into wild namely assign css vars to the tokens atm its mostly colors for the alpha test.

Please do report any issues to GH with syntax themes, this feedback is whats needed to get this resolved.

**Comment by jack1142** [comment]
Not so long ago I tried reaching out to GitHub on Twitter since the feedback sent through their private feedback form didn't get any response. I tried to explain the issues as detailed as possible, but I can only really talk about Python's syntax coloring as that's what I'm using most of the time.
 
Maybe someone will want to join with their own feedback or just share it so I hope people won't mind me sending a link here:
https://twitter.com/JakubKuczys/status/1337562630391812098

**Comment by the-j0k3r** [comment]
For contact I suggest the new and probably the right place https://support.github.com/tickets/personal/0
